configuration/parse.go

```diff
@@ -30,201 +31,176 @@ import (
 	"github.com/sonatype-nexus-community/nancy/audit"
 	. "github.com/sonatype-nexus-community/nancy/logger"
 	"github.com/sonatype-nexus-community/nancy/types"
+	"golang.org/x/crypto/ssh/terminal"
 	"gopkg.in/yaml.v2"
 )
 
-type Configuration struct {
-	UseStdIn   bool
-	Help       bool
-	NoColor    bool
-	Quiet      bool
-	Version    bool
-	CleanCache bool
-	CveList    types.CveListFlag
-	Path       string
-	Formatter  logrus.Formatter
-	Info       bool
-	Debug      bool
-	Trace      bool
-	Username   string `yaml:"Username"`
-	Token      string `yaml:"Token"`
-}
-
-type IqConfiguration struct {
+// Config is a struct for loading OSS Index or Nexus IQ Server config
+type Config struct {
+	Type        string
 	Help        bool
 	Version     bool
-	User        string
-	Token       string
 	Stage       string
 	Application string
 	Server      string
 	MaxRetries  int
 	Info        bool
 	Debug       bool
 	Trace       bool
+	UseStdIn    bool
+	NoColor     bool
+	Quiet       bool
+	CleanCache  bool
+	CveList     types.CveListFlag
+	Path        string
+	Formatter   logrus.Formatter
+	Username    string `yaml:"Username"`
+	Token       string `yaml:"Token"`
 }
```

Comment:
So I like the idea of the singular config object but not a fan of all properties jammed in there without knowing what is what. What if we use composition here and at least make a clearer distinction on what properties are for what and what is shared. 

```
type Config struct {
	Help       bool		
        Type        string
	NoColor    bool	
	Quiet      bool	
	Version    bool	
	CleanCache bool	
	CveList    types.CveListFlag	
	Path       string	
	Info       bool	
	Debug      bool	
	Trace      bool	
	Username   string `yaml:"Username"`	
	Token      string `yaml:"Token"`	
        IqConfiguration
        OssIndexConfiguration
}	

type IqConfiguration struct {	
	Stage       string
	Application string
	Server      string
	MaxRetries  int
}

type OssIndexConfiguration struct {	
	Formatter  logrus.Formatter	
}
```

Reply:
That's great! I may actually split the username and token into those structs too, because they are fully distinct.

Reply:
I think I got part of this in: 37493b2

Reply:
I'll move the other fields in, just gotta figure out if `go-yaml` will let me use `omitempty` to keep them from being persisted when we write to the file system (none of them are really global config, more like single use config)

Reply:
I think I finished up with moving stuff, here: 07117d3 (CleanCache could be thought of as just OSS Index specific, but I'm on the fence)

Reply:
So I'm still wrapping my head around all this, so ignore if this is crazy talk, but this reminded me of config code from the nfpm project (guts of goreleaser). They used a sort of "base" struct (named [Info](https://github.com/goreleaser/nfpm/blob/master/nfpm.go#L108)) to hold common config, and then used an [Overridables](https://github.com/goreleaser/nfpm/blob/master/nfpm.go#L129) struct for sub-type specific stuff. Could go deeper with type specific structs for [Rpm](https://github.com/goreleaser/nfpm/blob/master/nfpm.go#L145), [Deb](https://github.com/goreleaser/nfpm/blob/master/nfpm.go#L151). The secret sauce is how these get merged together via: [mergo](https://github.com/imdario/mergo). See how nfpm uses the lib [here](https://github.com/goreleaser/nfpm/blob/master/nfpm.go#L83) and [also here](https://github.com/goreleaser/nfpm/blob/master/nfpm.go#L91) 

(I wad [bitten](https://github.com/goreleaser/nfpm/pull/126#issuecomment-611721607) when I did the structs wrong, but it does seem pretty powerful)